### ISSUE TITLE
Implement File Size Limitation for AS2 Inbound Messages with Compressed Data

### ISSUE THREAD

**Issue by vamshikrishna45** [abhishek-ram/django-pyas2#120]
I've set file size limits for outbound AS2 messages, but I'm facing a challenge with inbound messages due to compression. Since the data is compressed, it's hard to set an accurate size limit before decompression. Any suggestions on how to handle this efficiently?

**Comment by chadgates** [comment]
For me, this setting from Django works - did you have issues with that ? 
https://docs.djangoproject.com/en/4.2/ref/settings/#data-upload-max-memory-size

**Comment by vamshikrishna45** [comment]
I encountered an issue where I need to reject incoming AS2 messages if
their size exceeds 500MB, but the file size limit should be configurable. I
initially tried handling this with middleware and methods in `views.py`.
However, due to the compression of incoming files, Django cannot determine
the exact file size. Could you suggest alternative solutions for managing
this?

**Comment by chadgates** [comment]
@vamshikrishna45 : yes, the one I mentioned above. This should resolve your issue - it is standard Django functionality.

**Comment by vamshikrishna45** [comment]
Thanks, I'll check it out.

**Comment by vamshikrishna45** [comment]
Hi,

I’m encountering an issue when retrying failed AS2 messages. Upon retry, a
new Message-ID is generated for the receiver, while the sender retains the
original Message-ID. This inconsistency occurs during the as2message.build()
process, which seems to be causing the generation of a new Message-ID.

Is there a way to maintain the same Message-ID for both the sender and
receiver during retries, ensuring that no new Message-ID is generated?

Thank you
